### PR TITLE
Add copy constuctor to MatrixFree::AdditionalData

### PR DESCRIPTION
**Pull request by peterrum**
follow-up to #8919

FYI @kronbichler

### CODE REVIEW

File: include/deal.II/matrix_free/matrix_free.h [hunk 1]
```diff
@@ -242,6 +242,30 @@ class MatrixFree : public Subscriptor
...
+      , mapping_update_flags_faces_by_cells(
+          other.mapping_update_flags_faces_by_cells)
+      , mg_level(other.mg_level)
+      , level_mg_handler(this->mg_level)
+      , store_plain_indices(other.store_plain_indices)
+      , initialize_indices(other.initialize_indices)
+      , initialize_mapping(other.initialize_mapping)
+      , overlap_communication_computation(
+          other.overlap_communication_computation)
+      , hold_all_faces_to_owned_cells(other.hold_all_faces_to_owned_cells)
```
Comment (kronbichler) [inline include/deal.II/matrix_free/matrix_free.h]: Did you forget `cell_vectorization_category`?

Comment (peterrum) [reply include/deal.II/matrix_free/matrix_free.h]: Fixed.

**Comment by kronbichler** [comment]
Thanks!

**Pull request merged**